project1/thread_pool/task.h

```diff
@@ -0,0 +1,60 @@
+#ifndef PROJECT1_TASK_H
+#define PROJECT1_TASK_H
+
+#include <cstdint>
+
+namespace thread_pool {
+
+/**
+ * @brief Represents a task that can be run in a tread pool.
+ */
+class Task {
+ public:
+  /// Represents an opaque identifier for a task.
+  using Handle = uint32_t;
```

Comment:
You're mapping the keyword `Handle` to the datatype `uint32_t` ? 

Is this for readability purposes only?

Reply:
Yes. Also because `Handle` is really just an opaque identifier and it doesn't really matter what type it is, as long as it's hashable.